src/libopensc/card-piv.c

```diff
@@ -880,11 +878,14 @@ static int piv_read_obj_from_file(sc_card_t * card, char * filename,
 		r =  SC_ERROR_DATA_OBJECT_NOT_FOUND;
 		goto err;
 	}
+	int r_tag;
+
 	body = tagbuf;
-	if (sc_asn1_read_tag(&body, 0xfffff, &cla_out, &tag_out, &bodylen) != SC_SUCCESS
+	r_tag = sc_asn1_read_tag(&body, len, &cla_out, &tag_out, &bodylen);
+	if ((r_tag != SC_SUCCESS && r_tag != SC_ERROR_ASN1_END_OF_CONTENTS)
 			|| body == NULL) {
 		sc_log(card->ctx, "DER problem");
-		r = SC_ERROR_INVALID_ASN1_OBJECT;
+		r = SC_ERROR_FILE_NOT_FOUND;
 		goto err;
 	}
 	rbuflen = body - tagbuf + bodylen;
```

Comment:
If I'm not mistaken then `bodylen` may not correctly initialized if `r_tag == SC_ERROR_ASN1_END_OF_CONTENTS`. The simplest would be to **not** try to cope with `SC_ERROR_ASN1_END_OF_CONTENTS`.

Reply:
At this point card-piv.c has only read the first few bytes, enough to read the tag and length. 
The point is to get the size of the full object into `rbuflen` so a buffer can be allocated then do a secondGET_DATA command to read the full object.

Reply:
I see, OK